File: frontend/src/types/meeting.ts
export enum MeetingType {
  Planning = 'Planning',
  Review = 'Review',
  Standup = 'Standup',
  Retrospective = 'Retrospective',
  OneOnOne = 'OneOnOne',
  Other = 'Other',
}

export enum MeetingStatus {
  Scheduled = 'Scheduled',
  InProgress = 'InProgress',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
}

export enum ParticipantStatus {
  Invited = 'Invited',
  Accepted = 'Accepted',
  Declined = 'Declined',
  Attended = 'Attended',
  Absent = 'Absent',
}

export interface Participant {
  id: string;
  userId: string;
  userName: string;
  userEmail: string;
  status: ParticipantStatus;
  invitedAt?: string;
  attendedAt?: string;
}

export interface AgendaItem {
  id: string;
  order: number;
  title: string;
  description?: string;
  estimatedDuration?: string;
  isCompleted: boolean;
}

export interface Decision {
  id: string;
  title: string;
  description: string;
  madeAt: string;
  madeById?: string;
  madeByName?: string;
  isImplemented: boolean;
}

export interface Task {
  id: string;
  meetingId: string;
  projectId?: string;
  assignedToId: string;
  assignedToName: string;
  title: string;
  description?: string;
  status: TaskStatus;
  priority: TaskPriority;
  dueDate?: string;
  completedAt?: string;
}

export enum TaskStatus {
  Pending = 'Pending',
  InProgress = 'InProgress',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
}

export enum TaskPriority {
  Low = 'Low',
  Medium = 'Medium',
  High = 'High',
  Critical = 'Critical',
}

export interface Meeting {
  id: string;
  organizationId: string;
  projectId?: string;
  organizerId: string;
  title: string;
  description?: string;
  type: MeetingType;
  status: MeetingStatus;
  scheduledAt: string;
  duration: string;
  location?: string;
  meetingUrl?: string;
  suggestedAgenda?: string;
  finalAgenda?: string;
  summary?: string;
  createdAt: string;
  startedAt?: string;
  endedAt?: string;
  projectName?: string;
  organizerName: string;
  participants: Participant[];
  agendaItems: AgendaItem[];
  decisions: Decision[];
  tasks: Task[];
}

export interface CreateMeetingDto {
  organizationId: string;
  projectId?: string;
  title: string;
  description?: string;
  type: MeetingType;
  scheduledAt: string;
  duration: string;
  location?: string;
  meetingUrl?: string;
  participantIds: string[];
}

export interface UpdateMeetingDto {
  title?: string;
  description?: string;
  type?: MeetingType;
  scheduledAt?: string;
  duration?: string;
  location?: string;
  meetingUrl?: string;
  finalAgenda?: string;
}

File: backend/src/SmartMeetingManager.Domain/Entities/Meeting.cs
namespace SmartMeetingManager.Domain.Entities;

public enum MeetingType
{
    Planning,
    Review,
    Standup,
    Retrospective,
    OneOnOne,
    Other
}

public enum MeetingStatus
{
    Scheduled,
    InProgress,
    Completed,
    Cancelled
}

public class Meeting
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public Guid? ProjectId { get; set; }
    public Guid OrganizerId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public MeetingType Type { get; set; }
    public MeetingStatus Status { get; set; }
    public DateTime ScheduledAt { get; set; }
    public TimeSpan Duration { get; set; }
    public string? Location { get; set; }
    public string? MeetingUrl { get; set; }
    public string? SuggestedAgenda { get; set; } // IA-generated
    public string? FinalAgenda { get; set; }
    public string? Summary { get; set; } // IA-generated post-meeting
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? EndedAt { get; set; }
    
    // Navigation properties
    public virtual Organization Organization { get; set; } = null!;
    public virtual Project? Project { get; set; }
    public virtual User Organizer { get; set; } = null!;
    public virtual ICollection<MeetingParticipant> Participants { get; set; } = new List<MeetingParticipant>();
    public virtual ICollection<AgendaItem> AgendaItems { get; set; } = new List<AgendaItem>();
    public virtual ICollection<Decision> Decisions { get; set; } = new List<Decision>();
    public virtual ICollection<Task> Tasks { get; set; } = new List<Task>();
    public virtual ICollection<Transcript> Transcripts { get; set; } = new List<Transcript>();
}